### PR TITLE
Hotbackup blockage after DBServer restart

### PR DESCRIPTION
**Pull request by neunhoef**
Add proper cleanup if a dbserver is restarted, shut down or crashes
during a hotbackup upload or download job. In this case we need to
detect this in the Supervision by looking at the rebootId, report
the upload or download job for this DBServer as failed, and clear the
lock for this operation and its destination.

The problem was that no rebootId was saved with the job, no cleanup
routine existed and the destination lock could remain, which would
prevent successful retries by the operator in production.

### Scope & Purpose

*(Please describe the changes in this PR for reviewers - **mandatory**)*

- [*] :hankey: Bugfix (requires CHANGELOG entry)
- [*] :book: CHANGELOG entry made

#### Backports:

- [*] Backports required for: 3.7
- [*] Forwardports required for: 3.9, devel

#### Related Information

*(Please reference tickets / specification / other PRs etc)*

- [*] GitHub issue / Jira ticket number: https://arangodb.atlassian.net/browse/BTS-652

### Testing & Verification

*(Please pick either of the following options)*

- [*] The behavior in this PR was *manually tested*
- [*] This PR adds tests that were used to verify all changes:
  - [*] Added new C++ **Unit tests**

### CODE REVIEW

**Review by kvahed — APPROVED** [review]
LGTM

File: arangod/Agency/Supervision.cpp [hunk 1]
```diff
@@ -1781,6 +1786,56 @@ void arangodb::consensus::cleanupHotbackupTransferJobsFunctional(
...
+            envelope->add("op", VPackValue("set"));
+            envelope->add("new", VPackValue("FAILED"));
+          }
+          envelope->add(VPackValue(std::string(HOTBACKUP_TRANSFER_LOCKS) + lockLocation.first));
+          {
+            VPackObjectBuilder guard(envelope.get());
+            envelope->add("op", VPackValue("delete"));
+          }
+        }
+      }
```
[the remark this answers is not in the document]
Comment (neunhoef) [reply arangod/Agency/Supervision.cpp:1852]: Done.

File: arangod/Agency/AgencyStrings.h [hunk 2]
```diff
@@ -50,6 +50,8 @@ constexpr char const* PLAN_DATABASES = "Plan/Databases/";
 constexpr char const* PLAN_VIEWS = "Plan/Views/";
 constexpr char const* PLAN_ANALYZERS = "Plan/Analyzers/";
 
+constexpr char const* HOTBACKUP_TRANSFER_LOCKS = "/Target/Hotbackup/Transfers/";
```
Comment (Robert-Stam) [inline arangod/Agency/AgencyStrings.h:53]: Hotbackup --> HotBackup?

Comment (neunhoef) [reply arangod/Agency/AgencyStrings.h:53]: No, we have unfortunately `/arango/Target/Hotbackup/Transfers` and `/arango/Target/HotBackup/TransferJobs` and in 3.8 I will not change this for backward compatibility reasons. In 3.9 and devel, I will change this and make it consistent. I will also use URLEncode in the agency paths.

**Review by kvahed — APPROVED** [review]
LGTM

**Pull request merged**